index.js

```diff
@@ -143,8 +139,8 @@ function SonicBoom (opts) {
           }, BUSY_WRITE_TIMEOUT)
         }
       } else {
-        this._writing = false
-
+        this._len -= this._writingBuf.length
+        this._writingBuf = ''
```

Comment:
The above two lines are not correct. For the "recoverable error" test to pass, you would need to unshift this inside `._bufs`.

Reply:
But this is in the unrecoverable path?

Reply:
It's potentially recoverable. The file might be reopen or possibly the file destination can be changed.

Reply:
I'm not sure I understand the semantics here. This way all errors will always be retried until they succeed. That sounds weird.

Reply:
The only different here between the `EAGAIN` path and this path is that the `EGAIN` path will retry in a loop while the other path will retry whenever `write` is called again... which essentially is the same thing... the loop just happens in different places.

Reply:
In this case the loop is user-controlled. They will listen to `'error'` if they want to. In case of an error, they can reopen this without losing data.

EAGAIN is explicitly handled instead.